cmake/Modules/CodeCoverage.cmake

```diff
@@ -139,6 +139,60 @@ function(ADD_CODE_COVERAGE)
         set(COVERAGE_DIR $ENV{ROS_HOME})
     endif()
 
+    # Ensure that the include component uses the correct path for symlinked source directories
+    get_filename_component(REAL_SOURCE_DIR ${PROJECT_SOURCE_DIR} REALPATH)
+
+    # python omit and include flags
+    list(APPEND Coverage_EXCLUDES "${REAL_SOURCE_DIR}/test/*" "${REAL_SOURCE_DIR}/tests/*")
+    string(REPLACE ";" "," Coverage_EXCLUDES_STR "${Coverage_EXCLUDES}")
+    set(OMIT_FLAGS "--omit=\"${Coverage_EXCLUDES_STR}\"")
+    set(INCLUDE_FLAGS "--include=\"${REAL_SOURCE_DIR}/*\"")
+
+    # add target for non-test repo
+    if(NOT TARGET run_tests_${PROJECT_NAME})
+      add_custom_target(run_tests_${PROJECT_NAME})
+    endif()
+
+    if(NOT TARGET _run_tests_${PROJECT_NAME})
+      # create hidden meta target which depends on hidden test targets which depend on clean_test_results
+      add_custom_target(_run_tests_${PROJECT_NAME})
+      # run_tests depends on this hidden target hierarchy to clear test results before running all tests
+     add_dependencies(run_tests _run_tests_${PROJECT_NAME})
+    endif()
+
+    if(NOT DEFINED CATKIN_ENABLE_TESTING OR CATKIN_ENABLE_TESTING)
+      # create python base coverage directory
+      add_custom_target(
+        create_python_base_coverage_dir "${CMAKE_COMMAND}" "-E" "make_directory" ${PROJECT_BINARY_DIR}/python_base_coverage
+      )
+
+      # create python base coverage report
+      add_custom_target(run_tests_${PROJECT_NAME}_python_base_coverage_report
+        COMMAND rosrun code_coverage generate_base_coverage.py ${PROJECT_SOURCE_DIR} --output ${PROJECT_BINARY_DIR}/python_base_coverage
```

Comment:
Could you add a comment about why we need to run this command?

Reply:
I add the comment.